packages/i18nify-js/src/modules/phoneNumber/getDialCodeByCountryCode.ts

```diff
@@ -0,0 +1,29 @@
+import { withErrorBoundary } from '../../common/errorBoundary';
+import getDialCodes from './getDialCodes';
+
+/**
+ * Retrieves the dial code for a specified country code.
+ *
+ * @param countryCode The country code for which the dial code is to be retrieved.
+ *                    It must be a key of the object returned by the `getDialCodes` function.
+ * @returns The corresponding dial code as a string.
+ * @throws An error if the provided country code is not found in the dial code mapping.
+ */
+const getDialCodeByCountryCode = (
+  countryCode: keyof ReturnType<typeof getDialCodes>,
+): string => {
+  // Get the mapping of all country codes to their respective dial codes
+  const dialCodeForAllCountries = getDialCodes();
```

Comment:
question: why don't you just call this out side, and use it when it is necessary

Reply:
What additional advantage will we get if `getDialCodes()` is called outside?